src/main/java/health/ere/ps/service/idp/BearerTokenService.java

```diff
@@ -39,13 +45,27 @@ public class BearerTokenService {
 
 
     @PostConstruct
-    public void init() {
+    public void init() throws Exception {
+        singleThreadPool.submit(() -> {
```

Comment:
please use a plain thread (no ExecutorService needed) keeping the threadpool open is a mini memory leak

Reply:
Fixed